dask/distributed/worker.py

```diff
@@ -534,10 +560,27 @@ def heart(self, pulse=5):
         """Send a message to scheduler at a given interval"""
         while self.status != 'closed':
             header = {'function': 'heartbeat'}
-            payload = {}
+            payload = {'pid': self.pid}
             self.send_to_scheduler(header, payload)
             self._heartbeat_thread.event.wait(pulse)
 
+    def worker_death(self, header, payload):
+        """
+        A worker died, check no queues are blocking waiting for data from the
+        worker.
+        """
+        with logerrors():
+            loads = header.get('loads', pickle.loads)
+            payload = loads(payload)
+            removed_workers = payload['removed']
+            for w in removed_workers:
+                for queue, keys in self.queues_by_worker[w].items():
+                    for k in keys:
```

Comment:
Eh, for imperative stuff like this I tend to prefer for loops.  This looks fine to me.